addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1467,18 +1467,18 @@ var SnippetsMenu = Widget.extend({
         if ($snippet && !$snippet.is(':visible')) {
             return;
         }
+        // Take the first parent of the provided DOM (or itself) which
+        // should have an associated snippet editor.
+        if ($snippet && $snippet.length) {
+            $snippet = globalSelector.closest($snippet);
+        }
```

Comment:
Add a comment to say it is important to do that before the mutex call otherwise we might break that again